src/main/kotlin/org/octopusden/octopus/oc/template/plugins/gradle/service/OcTemplateService.kt

```
@@ -263,6 +265,49 @@ abstract class OcTemplateService @Inject constructor(
+        val lastSeenPhase = mutableMapOf<String, String>()
+
+        logger.info("Waiting for pod(s) with prefix '$deploymentPrefix-$serviceName' to terminate (phase=Succeeded)...")
+
+        while (counter++ < attempts) {
+            Thread.sleep(period)
+            updateCreatedResources()
+            startLogStreaming()
+
+            val checkResult = checkPodAvailability(consecutiveNoPodChecks, seenAnyPod, TERMINATION_CONTEXT_LABEL)
+            if (checkResult.shouldExit) {
+                throw Exception("No pods observed for '$deploymentPrefix-$serviceName'; cannot verify completion")
```

detekt (1.23.8)

[warning] Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)